server/controllers/prepMaterialTeacherController.js

```diff
@@ -0,0 +1,146 @@
+var formidable = require("formidable");
+const fs = require("fs");
+var dir = __dirname + "/../uploads/qnaStudents/";
+const { TeacherPrepMaterial } = require("../models/prepMaterialTeacher.model");
+const { mongo } = require("mongoose");
+
+const path = require("path");
+const responseHelpers = require("../helpers/response.helpers");
+
+// API to post Practice Quiz Test from teacher side login  :
+exports.postCreateQuiz = async (req, res, next) => {
+  if (req.user.role == "teacher") {
```

Comment:
what if `req.user` is not available?
compare with deep equality

Reply:
sure .. let me make the fixes and I will push by EOD today .